src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,29 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String getSnapshotName(SnapshotRequestModel model) {
+    SnapshotRequestContentsModel contentsModel = model.getContents().get(0);
+    if (contentsModel.getMode() == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      SnapshotAccessRequestResponse snapshotAccessRequestResponse =
+          snapshotRequestDao.getById(contentsModel.getRequestIdSpec().getSnapshotRequestId());
+      String dashesAndSpacesRegex = "[- ]+";
+      String nonAlphaNumericRegex = "\\W";
+
+      return org.springframework.util.StringUtils.trimTrailingCharacter(
```

Comment:
I'm not sure why you use trimTrailing here, as I think trailing `_`s are allowed. Although since the end of the name is the UUID, we know it won't end with that character anyway.